src/dune_pkg/opam_solver.ml

```diff
@@ -482,5 +482,40 @@ let solve_lock_dir solver_env version_preference (repo, repo_id) ~local_packages
       | Ok pkgs_by_name ->
         Lock_dir.create_latest_version pkgs_by_name ~ocaml:None ~repo_id
     in
-    summary, lock_dir)
+    let files =
+      let read path =
+        let rec read acc path =
+          match Path.readdir_unsorted_with_kinds path with
+          | Ok files ->
+            List.concat_map files ~f:(fun (filename, kind) ->
+              match (kind : Unix.file_kind) with
+              | S_REG -> [ acc @ [ filename ] ]
+              | S_DIR -> read (acc @ [ filename ]) (Path.relative path filename)
+              | _ -> [])
+          | Error _ ->
+            (* TODO *)
```

Comment:
Not sure what to do in this situation either.

Reply:
This situation happens when we are unable to read a directory. It returns a Unix error so I guess the best we can do is raise it.

Reply:
I've added a test demonstrating what happens if we get a unix error whilst trying to read the `files/` directory.